File: 백준/Gold/22862. 가장 긴 짝수 연속한 부분 수열 （large）/가장 긴 짝수 연속한 부분 수열 （large）.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());
        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int s = 0, e = 0, result = 0, sum = 0, ability = k;

        while (e < n) {
            if (arr[e] % 2 == 0) {
                sum++;
                result = Math.max(result, sum);
                e++;
            } else {
                if (ability > 0) {
                    ability--;
                    e++;
                } else {
                    if (arr[s] % 2 != 0) {
                        ability++;
                    } else {
                        sum--;
                    }
                    s++;
                }
            }
        }
        System.out.println(result);
    }
}

File: 백준/Gold/22862. 가장 긴 짝수 연속한 부분 수열 （large）/가장 긴 짝수 연속한 부분 수열 （large）.py
import sys
input = lambda: sys.stdin.readline().strip()
n,k = map(int,input().split())
arr = list(map(int,input().split()))
i=0
while i<n and arr[i]:
    i+=1
    

s,e=0,0
result = 0
sum=0
ability = k

while e<n:
    # print(s,e,sum, ability)
    if arr[e]%2==0:
        sum+=1
        result = max(result,sum)
        e+=1
    else:
        if ability:
            ability-=1
            e+=1
        else:
            if arr[s]%2:
                ability+=1
            else:
                sum-=1
            s+=1    
        
print(result)